# HG changeset patch
# User MJP
# Date 1390488664 -3600
#      Thu Jan 23 15:51:04 2014 +0100
# Node ID 31e8f8119e380ad69e62be3215eb606681600e4c
# Parent  a49fe392596c985ed10feb7aae94851817ad4ef4
No wait cursor for a threaded autosave

diff -r a49fe392596c -r 31e8f8119e38 src/saveload/saveload.cpp
--- a/src/saveload/saveload.cpp	Thu Jan 23 15:51:00 2014 +0100
+++ b/src/saveload/saveload.cpp	Thu Jan 23 15:51:04 2014 +0100
@@ -2390,11 +2390,12 @@
  * and set locks on saveload. Also turn off fast-forward cause with that
  * saving takes Aaaaages
  */
-static void SaveFileStart()
+static void SaveFileStart(bool threaded)
 {
 	_sl.ff_state = _fast_forward;
 	_fast_forward = 0;
-	if (_cursor.sprite == SPR_CURSOR_MOUSE) SetMouseCursor(SPR_CURSOR_ZZZ, PAL_NONE);
+	if (_do_autosave && threaded) {
+	} else if (_cursor.sprite == SPR_CURSOR_MOUSE) SetMouseCursor(SPR_CURSOR_ZZZ, PAL_NONE);
 
 	InvalidateWindowData(WC_STATUS_BAR, 0, SBI_SAVELOAD_START);
 	_sl.saveinprogress = true;
@@ -2517,7 +2518,7 @@
 	SaveViewportBeforeSaveGame();
 	SlSaveChunks();
 
-	SaveFileStart();
+	SaveFileStart(threaded);
 	if (!threaded || !ThreadObject::New(&SaveFileToDiskThread, NULL, &_save_thread)) {
 		if (threaded) DEBUG(sl, 1, "Cannot create savegame thread, reverting to single-threaded mode...");
 
